### PR TITLE
Add some common account bits and some getOpts -> GetOpts changes

### PR DESCRIPTION
**Pull request by jefferai**
This is a precursor to another upcoming PR that will make use of these changes, but it makes that PR more manageable.

### CODE REVIEW

File: internal/authtoken/options.go [hunk 1]
```diff
@@ -91,3 +95,21 @@ func WithPublicId(id string) Option {
...
+}
+
+// WithIamOptions allows passing through options for the IAM package. This is
+// useful for things like tests where you may use testing helper functions in
+// this package but they also create IAM resources.
+func WithIamOptions(with ...iam.Option) Option {
+	return func(o *options) {
+		o.withIamOptions = with
+	}
+}
```
Comment (johanbrandhorst) [inline internal/authtoken/options.go:115]: Should these go in `testing.go`?

File: internal/daemon/controller/handlers/targets/registry.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ type registryEntry struct {
 }
 
 type registry struct {
-	sync.Map
+	*sync.Map
```
Comment (johanbrandhorst) [inline internal/daemon/controller/handlers/targets/registry.go:42]: 😱 was this buggy the whole time?

> A Map must not be copied after first use

https://pkg.go.dev/sync#Map

**Review by johanbrandhorst — APPROVED** [review]

File: internal/daemon/controller/handlers/targets/registry.go (hunk 2 above)
Comment (jefferai) [reply internal/daemon/controller/handlers/targets/registry.go:42]: Yep! It was!

File: internal/authtoken/options.go (hunk 1 above)
Comment (jefferai) [reply internal/authtoken/options.go:115]: The reason I didn't -- and maybe this is wrong -- is that they do go into normal non-testing functions, even if they're mostly used for tests. So I didn't want to have e.g. `NewAccount(..., ..., ..., password.TestOptions...)`. If these were only ever used in actual testing functions I would definitely put them in testing.go and label them e.g. `WithTestIamOption` and `WithTestPasswordOptions`.

WDYT?

Comment (jefferai) [reply internal/authtoken/options.go:115]: We can always move it later; resolving for now.

**Pull request merged**